### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by steve-downey**
Improve the CI workflow, building with clang 17, 18, and 19, against libc++, and  with GCC  13 and 14 using Ubuntu 24.04 as the base OS.

### CODE REVIEW

**Review by neatudarius — APPROVED** [review]

File: etc/clang-flags.cmake [hunk 1]
```diff
@@ -3,7 +3,7 @@ include_guard(GLOBAL)
 set(CMAKE_CXX_STANDARD 23)
```
Comment (JeffGarland) [inline etc/clang-flags.cmake:3]: remind me -- is there something here that requires 23, or could it be 20?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,76 +1,121 @@
-name: ci
+name: Test
```
Comment (JeffGarland) [inline .github/workflows/ci.yml:1]: I'm no expert on these github workflows now - @bretbrownjr any comments?  Specifically I'm wondering about non-Linux platforms.  Also, @steve-downey did this come from our template or do we need to update?

**Review by JeffGarland — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (neatudarius) [reply .github/workflows/ci.yml:1]: @steve-downey , is there a way to run these flows on local environment? e.g. git API to span a Docker for every job, but on my local setup. I am just curious if I am doing a change, if I can run all jobs on my MAC OS / Ubuntu.

Comment (steve-downey) [reply .github/workflows/ci.yml:1]: I have not found a way to simulate any of this locally. I've ended up enabling actions on PRs and repeated force-pushes to a branch to trigger a run.

Comment (steve-downey) [reply .github/workflows/ci.yml:1]: I took a bunch of it from https://github.com/beman-project/example/pull/6. I also have some possible updates to deal with Node version warnings over here: https://github.com/steve-downey/lambda-fun/pull/2/files

File: etc/clang-flags.cmake (hunk 1 above)
Comment (steve-downey) [reply etc/clang-flags.cmake:3]: I'll check. I believe at one point I was looking at `deducing this` but then didn't.

**Comment by steve-downey** [comment]
Merging so as to base further work on working CI

**Pull request merged**